apis/core/v1beta1/featureflaginprocessconfiguration_types.go

```diff
@@ -0,0 +1,195 @@
+type FeatureFlagInProcessConfigurationStatus struct {
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// FeatureFlagInProcessConfiguration is the Schema for the featureflaginprocesconfigurations API
+type FeatureFlagInProcessConfiguration struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   FeatureFlagInProcessConfigurationSpec   `json:"spec,omitempty"`
+	Status FeatureFlagInProcessConfigurationStatus `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// FeatureFlagInProcessConfigurationList contains a list of FeatureFlagInProcessConfiguration
+type FeatureFlagInProcessConfigurationList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []FeatureFlagInProcessConfiguration `json:"items"`
+}
+
+func init() {
+	SchemeBuilder.Register(&FeatureFlagInProcessConfiguration{}, &FeatureFlagInProcessConfigurationList{})
+}
+
+func (fc *FeatureFlagInProcessConfigurationSpec) Merge(new *FeatureFlagInProcessConfigurationSpec) {
+	if new == nil {
+		return
+	}
+	if len(new.EnvVars) != 0 {
+		fc.EnvVars = append(fc.EnvVars, new.EnvVars...)
+		fc.EnvVars = common.RemoveDuplicateEnvVars(fc.EnvVars)
+	}
+
+	if new.Port != common.DefaultPort {
+		fc.Port = new.Port
+	}
+	if new.SocketPath != "" {
+		fc.SocketPath = new.SocketPath
+	}
+	if new.Host != common.DefaultHost {
+		fc.Host = new.Host
+	}
+	if new.EnvVarPrefix != common.DefaultEnvVarPrefix {
+		fc.EnvVarPrefix = new.EnvVarPrefix
+	}
+	if new.OfflineFlagSourcePath != "" {
+		fc.OfflineFlagSourcePath = new.OfflineFlagSourcePath
+	}
+	if new.Selector != "" {
+		fc.Selector = new.Selector
+	}
+	if new.Cache != common.DefaultCache {
+		fc.Cache = new.Cache
+	}
+	if new.CacheMaxSize != int(common.DefaultCacheMaxSize) {
+		fc.CacheMaxSize = new.CacheMaxSize
+	}
+	if new.TLS != common.DefaultTLS {
+		fc.TLS = new.TLS
+	}
+}
+
+func (fc *FeatureFlagInProcessConfigurationSpec) ToEnvVars() []corev1.EnvVar {
+	envs := []corev1.EnvVar{}
+
+	for _, envVar := range fc.EnvVars {
+		envs = append(envs, corev1.EnvVar{
+			Name:  common.EnvVarKey(fc.EnvVarPrefix, envVar.Name),
+			Value: envVar.Value,
+		})
+	}
+
+	envs = append(envs, corev1.EnvVar{
+		Name:  common.EnvVarKey(fc.EnvVarPrefix, common.HostEnvVar),
+		Value: fc.Host,
+	})
+
+	envs = append(envs, corev1.EnvVar{
+		Name:  common.EnvVarKey(fc.EnvVarPrefix, common.PortEnvVar),
+		Value: fmt.Sprintf("%d", fc.Port),
+	})
+
+	envs = append(envs, corev1.EnvVar{
+		Name:  common.EnvVarKey(fc.EnvVarPrefix, common.TLSEnvVar),
+		Value: fmt.Sprintf("%t", fc.TLS),
+	})
+
+	envs = append(envs, corev1.EnvVar{
+		Name:  common.EnvVarKey(fc.EnvVarPrefix, common.SocketPathEnvVar),
+		Value: fc.SocketPath,
+	})
+
+	envs = append(envs, corev1.EnvVar{
+		Name:  common.EnvVarKey(fc.EnvVarPrefix, common.OfflineFlagSourcePathEnvVar),
+		Value: fc.OfflineFlagSourcePath,
+	})
```

Comment:
Should we support this? @Kavindu-Dodan @beeme1mr 

This is more of a non-prod thing, I think, right?

Reply:
yeah non-prod but a valid use-case so things can work offline